engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java

```diff
@@ -269,6 +272,31 @@ public PageConfigurationDto getPageConfiguration(String pageCode, String status)
         return pageConfigurationDto;
     }
 
+    @Override
+    public PageConfigurationDto restorePageConfiguration(String pageCode) {
+        try {
+            IPage pageD = this.loadPage(pageCode, STATUS_DRAFT);
+            if (null == pageD) {
+                throw new RestRourceNotFoundException(ERRCODE_PAGE_NOT_FOUND, "page", pageCode);
+            }
+            IPage pageO = this.loadPage(pageCode, STATUS_ONLINE);
+            if (null == pageO) {
+                DataBinder binder = new DataBinder(pageCode);
+                BindingResult bindingResult = binder.getBindingResult();
+                bindingResult.reject(ERRCODE_STATUS_INVALID, new String[]{pageCode}, "page.status.invalid");
+                throw new ValidationGenericException(bindingResult);
+            }
+            ((Page) pageD).setMetadata(pageO.getMetadata());
```

Comment:
Why is this cast necessary? Should we just update the interface?

Reply:
That interface does not have setters (only getters).

Reply:
Let's update the interface to expose these methods. Casting to the concrete implementation here isn't good.